### PR TITLE
Studio: Ensure the  `Disconnect` button is disabled only when the particular site is syncing

### PR DESCRIPTION
**Pull request by ivan-ottinger**
## Related issues

- Resolves https://github.com/Automattic/dotcom-forge/issues/9993. 

## Proposed Changes

- Ensure the  `Disconnect` button is disabled only when the particular site is syncing

Markup on 2024-11-28 at 15:57:35

## Testing Instructions

1. Check out the PR and build the app with `STUDIO_SITE_SYNC=true npm start`.
2. Connect one of your local sites with multiple WPCOM sites.
3. Initiate pull or push operation and while it is in progress, try to disconnect one of the other connected WPCOM sites.
4. It should work correctly.
5. It should not be possible to disconnect the site that is syncing.
6. The above behavior should work the same when syncing staging sites as well.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Review by katinthehatsite — APPROVED** [review]
The changes look good to me and work as expected on my end 👍 
Nice work!

**Comment by wojtekn** [comment]
@ivan-ottinger I tested it, but I was able to disconnect site I'm pulling into, too:

Screenshot 2024-11-29 at 09 36 04

**Comment by ivan-ottinger** [comment]
> @ivan-ottinger I tested it, but I was able to disconnect site I'm pulling into, too:

That's odd. I cannot reproduce that issue. 🤔 To make sure I understand correctly: In your case it was possible to click on the `Disconnect` link of a site that is currently pulling from WPCOM (on the Sync tab)?

**Comment by wojtekn** [comment]
@ivan-ottinger yes, I reproduced it again, but now it didn't crash:

https://github.com/user-attachments/assets/2d3d5c93-93ce-4bf9-be72-1a2396121549

**Comment by ivan-ottinger** [comment]
> @ivan-ottinger yes, I reproduced it again, but now it didn't crash:

Thanks for sharing the screen recording. I can reproduce the issue now: it is happening when pulling from (or pushing to) a staging site only. I will take a look what's the cause.

**Comment by ivan-ottinger** [comment]
> > @ivan-ottinger yes, I reproduced it again, but now it didn't crash:
>
> Thanks for sharing the screen recording. I can reproduce the issue now: it is happening when pulling from (or pushing to) a staging site only. I will take a look what's the cause.

I have addressed the issue in https://github.com/Automattic/studio/pull/706/commits/26728c034bc3f04ca3489081082143c20a5846c8 and the PR is now ready for another review round. 🙂

File: src/components/sync-connected-sites.tsx [hunk 1]
```diff
@@ -132,10 +132,14 @@ const SyncConnectedSitesSection = ( {
 
 	const mainSite = section.connectedSites.find( ( item ) => ! item.isStaging );
 	const hasConnectionErrors = mainSite?.syncSupport !== 'already-connected';
-	const sitePullState = getPullState( selectedSite.id, section.id );
-	const isPulling = sitePullState && isKeyPulling( sitePullState.status.key );
-	const sitePushState = getPushState( selectedSite.id, section.id );
-	const isPushing = sitePushState.isInProgress;
+	const isAnySectionSitePulling = section.connectedSites.some( ( site ) => {
```
Comment (wojtekn) [inline src/components/sync-connected-sites.tsx]: I think `isPulling` still makes sense as a simpler name.

**Review by wojtekn — APPROVED** [review]
Works well now.

File: src/components/sync-connected-sites.tsx (hunk 1 above)
Comment (ivan-ottinger) [reply src/components/sync-connected-sites.tsx]: Thanks for the feedback. I have updated the names in https://github.com/Automattic/studio/pull/706/commits/2d9451d3c5ab018b2879cfb455f7fd2fec7c00a1 and merged the PR.

**Pull request merged**